pkg/vm/stack_item.go

```diff
@@ -857,3 +857,78 @@ func (i *InteropItem) Convert(typ StackItemType) (StackItem, error) {
 func (i *InteropItem) MarshalJSON() ([]byte, error) {
 	return json.Marshal(i.value)
 }
+
+// PointerItem represents VM-level instruction pointer.
+type PointerItem struct {
+	pos    int
+	script []byte
+}
+
+// NewPointerItem returns new pointer on the specified position.
+func NewPointerItem(pos int, script []byte) *PointerItem {
+	return &PointerItem{
+		pos:    pos,
+		script: script,
+	}
+}
+
+// String implements StackItem interface.
+func (p *PointerItem) String() string {
+	return "Pointer"
+}
+
+// Value implements StackItem interface.
+func (p *PointerItem) Value() interface{} {
+	return p.pos
+}
+
+// Dup implements StackItem interface.
+func (p *PointerItem) Dup() StackItem {
+	return NewPointerItem(p.pos, p.script)
+}
+
+// Bool implements StackItem interface.
+func (p *PointerItem) Bool() bool {
+	return true
+}
+
+// TryBytes implements StackItem interface.
+func (p *PointerItem) TryBytes() ([]byte, error) {
+	return nil, errors.New("can't convert Pointer to ByteArray")
+}
+
+// TryInteger implements StackItem interface.
+func (p *PointerItem) TryInteger() (*big.Int, error) {
+	return nil, errors.New("can't convert Pointer to Integer")
+}
+
+// Equals implements StackItem interface.
+func (p *PointerItem) Equals(s StackItem) bool {
+	if p == s {
+		return true
+	}
+	ptr, ok := s.(*PointerItem)
+	return ok && p.pos == ptr.pos && bytes.Equal(p.script, ptr.script)
+}
+
+// ToContractParameter implements StackItem interface.
+func (p *PointerItem) ToContractParameter(map[StackItem]bool) smartcontract.Parameter {
+	panic("implement me")
```

Comment:
Well, I agree, but `InteropItem` is also a valid contract parameter type. Pointer is internal only, so the best I have come up with is to use `Any` type.

Reply:
> `InteropItem` is also a valid contract parameter type

True, but at the moment our `smartcontract.Parameter` is only used for outside communication, like when receiving these parameters as a part of RPC call, or when answering to an RPC call. You don't need it when you're inside a VM. I think this function is more related to serializeability of the element, but at the same time it can't panic or throw an error.

> Pointer is internal only

It's not for long, see neo-project/neo#1629.